pkg/serverless/trace/trace.go

```diff
@@ -71,13 +69,11 @@ func (s *ServerlessTraceAgent) Start(enabled bool, loadConfig Load, executionCon
 			tc.Hostname = ""
 			tc.SynchronousFlushing = true
 			s.ta = agent.NewAgent(context, tc)
-			s.spanModifier = &spanModifier{}
-			if executionContext != nil {
-				s.spanModifier.coldStartSpanCreator = &ColdStartSpanCreator{
-					executionContext: executionContext,
-					traceAgent:       s,
-					createSpan:       &sync.Once{},
-				}
+			s.spanModifier = &spanModifier{
+				ColdStartSpanId: coldStartSpanId,
+			}
+			if lambdaSpanChan != nil {
+				s.spanModifier.lambdaSpanChan = lambdaSpanChan
```

Comment:
Is there an instance in which we wouldn't want the span chan to be set? Later on down the road a nil span chan could cause a panic. Might be worth it to consider making a new channel here in the case where the caller does not provide one.

Reply:
It's not set for `serverless-init` use cases, where there is no extension running (as it's used in other clouds that don't have extensions)

Reply:
Correct. And if the value of `lambdaSpanChan` that is passed into this method is `nil`, then it does not harm to set `s.spanModifier.lambdaSpanChan` to `nil`. What I mean si that I think you can safely do

```go
			s.spanModifier = &spanModifier{
				coldStartSpanId: coldStartSpanId,
                lambdaSpanChan:  lambdaSpanChan,
			}
```

But this is a nitpick and a completely unnecessary change, so feel free to continue to ignore it 😆